parquet-column/src/test/java/org/apache/parquet/schema/TestPrimitiveComparator.java

```diff
@@ -268,6 +271,36 @@ public void testBinaryAsSignedIntegerComparatorWithEquals() {
     }
   }
 
+  @Test
+  public void testFloat16Comparator() {
+    short[] valuesInAscendingOrder = {
+      (short) 0xfc00,
+      Float16.MIN_VALUE,
+      -Float16.MAX_VALUE,
+      (short) 0xc000,
+      -Float16.MIN_VALUE,
+      0,
+      Float16.MIN_VALUE,
+      (short) 0x7bff,
+      Float16.MAX_VALUE,
+      (short) 0x7c00};
+
+    for (int i = 0; i < valuesInAscendingOrder.length; ++i) {
+      for (int j = 0; j < valuesInAscendingOrder.length; ++j) {
+        short vi = valuesInAscendingOrder[i];
+        short vj = valuesInAscendingOrder[j];
+        ByteBuffer bbi = ByteBuffer.allocate(2).order(ByteOrder.LITTLE_ENDIAN);
+        bbi.putShort(vi).flip();
+        ByteBuffer bbj = ByteBuffer.allocate(2).order(ByteOrder.LITTLE_ENDIAN);
+        bbj.putShort(vj).flip();
+        float fi = Float16.toFloat(vi);
+        float fj = Float16.toFloat(vj);
+        assertEquals(Float.compare(fi, fj), BINARY_AS_FLOAT16_COMPARATOR.compare(
```

Comment:
If values are in the ascending order, we should also verify that if i < j, then fi < fj.